### PR TITLE
Implement typeref to demangle tree conversion for nested bound generics

### PR DESCRIPTION
**Pull request by augusto2112**

### CODE REVIEW

**Comment by augusto2112** [comment]
@swift-ci smoke test

**Review by adrian-prantl — APPROVED** [review]
Code looks good. Do we have test coverage for this in the validation tests?

File: stdlib/public/Reflection/TypeRef.cpp [hunk 1]
```diff
@@ -579,10 +579,37 @@ class DemanglingForTypeRef
...
+
+    // Remove all children.
+    nominalNode->removeChildAt(1);
+    nominalNode->removeChildAt(0);
+    
+    // Add the parent we just visited back in, followed by the identifier.
+    nominalNode->addChild(parentNode, Dem);
+    nominalNode->addChild(identifierNode, Dem);
     return genericNode;
+
```
Comment (mikeash) [inline stdlib/public/Reflection/TypeRef.cpp:612]: Stray blank line here.

**Review by mikeash — APPROVED** [review]
One teeny cosmetic comment, LGTM.

**Comment by augusto2112** [comment]
@swift-ci smoke test

**Comment by augusto2112** [comment]
I accidentally merged this on

**Comment by augusto2112** [comment]
I accidentally merged the correct version of this change here: https://github.com/apple/swift/pull/59262

**Pull request closed without merging**